src/mono/mono/mini/aot-compiler.c

```diff
@@ -4977,7 +4977,7 @@ add_full_aot_wrappers (MonoAotCompile *acfg)
 		add_method (acfg, get_runtime_invoke_sig (csig));
 
 		/* runtime-invoke used by finalizers */
-		add_method (acfg, get_runtime_invoke (acfg, get_method_nofail (mono_defaults.object_class, "Finalize", 0, 0), TRUE));
+		add_method (acfg, mono_marshal_get_runtime_invoke_full (get_method_nofail (mono_defaults.object_class, "GuardedFinalize", 0, 0), FALSE, TRUE));
```

Comment:
@lambdageek - I think I am precompiling `GuardedFinalize` around here.  (Although I am not sure)

Reply:
The signature for the `GuardedFinalize` is the same as for `Finalize`, 
If I understand correctly what `add_method (acfg, get_runtime_invoke_sig (csig));` does,  
then the `/* void runtime-invoke () [Finalize] */` should do what we need. Perhaps the comment needs to be changed.

Reply:
you're right, sorry. I'm not sure what the problem is